python/lsst/obs/base/filters.py

```diff
@@ -285,6 +184,117 @@ def defineFilter(self):
                 alias=sorted(aliases),
             )
 
-    def makeFilterLabel(self):
+    def makeFilterLabel(self) -> lsst.afw.image.FilterLabel:
```

Comment:
Indeed.  I wonder a bit whether the right solution is to to split obs_base into two pieces rather than try to ABC-ify the bits that have stack dependencies.